src/main/scala/contadamination/ContaminationModel.scala

```diff
@@ -0,0 +1,179 @@
+package contadamination
+
+import com.twitter.algebird.{BloomFilterMonoid, BloomFilter, BF}
+import contadamination.ContaminationModel.{ContaminationStats, FilterContext}
+import org.apache.spark.mllib.rdd.RDDFunctions
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.rdd.ADAMContext
+import org.bdgenomics.formats.avro.AlignmentRecord
+
+import scala.collection
+
+object ContaminationModel {
+
+  val defaultWinSize = 32
+  val defaultFpr = 0.1
+  val defaultSeed = 0
+
+  case class FilterContext(
+                name: String,     // Contig name
+                path: String,     // Reference path
+                numEntries: Int,  // Number of sliding windows
+                fpr: Double,      // False Positive Rate
+                bloomFilter: BF            // Bloom filter
+  )
+
+  case class ContaminationStats(
+                numReads: Int,    // Total number of reads checked
+                tn: Int,          // True Negatives (number of rejected reads)
+                rate: Double      // Contamination rate as the number of rejected reads over the total number of reads
+  )
+  
+  /**
+   * Build a ContaminationModel from a sequence of paths
+   * @param paths Sequence of FASTA files for the reference contigs
+   * @param winSize Sliding window size
+   * @param fpr False Positive Rate
+   * @param adamContext ADAM Context
+   * @return a ContaminationModel
+   */
+  def apply(paths: Seq[String], winSize: Int = defaultWinSize, fpr: Double = defaultFpr)(implicit adamContext: ADAMContext): ContaminationModel = {
+    paths.map(ContaminationModel(_, winSize, fpr)).reduce((x, y) => x ++ y)
+  }
+  
+  /**
+   * Build a ContaminationModel from a path
+   * @param path FASTA file with the reference contig
+   * @param winSize Sliding window size
+   * @param fpr False Positive Rate for the BloomFilter
+   * @param adamContext ADAM Context
+   * @return a ContaminationModel
+   */
+  def apply(path: String, winSize: Int = defaultWinSize, fpr: Double = defaultFpr)(implicit adamContext: ADAMContext): ContaminationModel = {
+    val fragments = adamContext.loadFasta(path, winSize).cache()
```

Comment:
I should have explained this better, this is my assumption here... We are going to check reads of a certain size, so our bloom filter should be build with fragments of this size, so I load the contigs in fragments of the same size as the reads, which simplifies quite a lot the building of the bloom filter, and the posterior checks.

Are you suggesting that the bloom filters should be build with sequences of smaller size than the reads and then check for all the sliding windows of a read of this smaller size ? then we would consider that a read is a contamination if neither of the sliding windows of it is contained in the bloom filters ?